scan.go

```diff
@@ -272,9 +265,53 @@ func scan(pkgPatterns []string, scanDir string) ([]*unit.SourceUnit, error) {
 			Ops:          map[string]*srclib.ToolRef{"depresolve": nil, "graph": nil},
 		})
 	}
-	if err := cmd.Wait(); err != nil {
+
+	return units, nil
+}
+
+func scanForPackages(dir string) ([]*build.Package, error) {
+	var pkgs []*build.Package
+
+	pkg, err := buildContext.ImportDir(dir, 0)
+	if _, isNoGoError := err.(*build.NoGoError); err != nil && !isNoGoError {
 		return nil, err
 	}
+	if err == nil {
+		pkgs = append(pkgs, pkg)
+	}
 
-	return units, nil
+	infos, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	for _, info := range infos {
+		name := info.Name()
+		fullPath := filepath.Join(dir, name)
+		if info.IsDir() && ((name[0] != '.' && name[0] != '_' && name != "testdata") || (strings.HasSuffix(fullPath, "Godeps/_workspace") && !config.SkipGodeps)) {
```

Comment:
> is necessary so the import paths are properly detected.

Oh, I see, you want the Godeps workspace Go packages to have an import path that includes their location inside the Godeps workspace (e.g., `github.com/sourcegraph/srclib-go/Godeps/_workspace/src/golang.org/x/tools/godoc/vfs/httpfs`). Then this approach makes sense.